flake.nix

```diff
@@ -0,0 +1,107 @@
+{
+  description = "cwe_checker finds vulnerable patterns in binary executables";
+
+  inputs = {
+    # use upstream once https://github.com/NixOS/nixpkgs/pull/140208 is accepted
+    # nixpkgs.url = "nixpkgs/nixos-21.05";
+    nixpkgs.url = "github:ilkecan/nixpkgs/nixos-21.05";
+    flake-utils.url = "github:numtide/flake-utils";
+    nix-filter.url = "github:numtide/nix-filter";
+
+    fenix = {
+      url = "github:nix-community/fenix";
+      inputs.nixpkgs.follows = "nixpkgs";
+    };
+
+    nix-utils = {
+      url = "git+https://git.sr.ht/~ilkecan/nix-utils";
+      inputs.nixpkgs.follows = "nixpkgs";
+    };
+  };
+
+  outputs = { self, nixpkgs, flake-utils, ... }@inputs:
+    let
+      inherit (builtins)
+        attrNames
+        attrValues
+      ;
+      inherit (nixpkgs.lib)
+        getAttrs
+        intersectLists
+      ;
+      inherit (flake-utils.lib)
+        defaultSystems
+        eachSystem
+      ;
+      nix-filter = inputs.nix-filter.lib;
+      nix-utils = inputs.nix-utils.lib;
+      inherit (nix-utils)
+        createOverlays
+        importCargoLock
+      ;
+
+      # ghidra-bin.meta.platforms
+      ghidraPlatforms = [ "x86_64-linux" "x86_64-darwin" ];
```

Comment:
ghidra derivation in Nixpkgs only supports these two platforms. I think this is a limitation of Nixpkgs rather than upstream project. But since `cwe_checker` depends on `ghidra`, I could only support these two platforms.